### PR TITLE
Add wait_for_any_object functionality to object_tree module

### PR DESCRIPTION
**Pull request by pawtom**
I created this PR to have a place to discuss two new functions:

```
def wait_for_any_object(
    object_names: list,
    timeout=squish.testSettings.waitForObjectTimeout / 1000,
    retry_delay: float = 0.5,
)

def wait_for_any_object_exists(
    object_names: list,
    timeout=squish.testSettings.waitForObjectTimeout / 1000,
    retry_delay: float = 0.5,
)
```

### CODE REVIEW

File: squape/object_tree.py [hunk 1]
```diff
@@ -269,3 +271,85 @@ def _get_object_reference(object_or_name: any) -> any:
...
+        for obj_name in object_names:
+            try:
+                return lookup_function(obj_name, 0)
+
+            except LookupError as e:
+                lookup_errors.add(str(e))
+
+        elapsed_time = time.time() - start_time
+        if elapsed_time > timeout:
+            break
```
Comment (Adakar) [inline squape/object_tree.py:293]: How about putting that into the while condition?

File: squape/object_tree.py [hunk 2]
```diff
@@ -269,3 +271,85 @@ def _get_object_reference(object_or_name: any) -> any:
...
+
+        squish.snooze(retry_delay)
+
+    raise LookupError(f"Objects not found: {lookup_errors}")
```
Comment (Adakar) [inline squape/object_tree.py:297]: It seems that you may end up with tons of lookup errors. Each object may cause different lookup errors over time, if you multiply it by the number of objects things can get crazy :P
Maybe we should report just one but mentioning all the objects that were provided in `object_names`?
I know that different objects may not be accessible due to a different reasons. How about adding DEBUG messages printing details of the **last** lookup error for each object?

**Review by Adakar — CHANGES_REQUESTED** [review]

File: squape/object_tree.py (hunk 1 above)
Comment (pawtom) [reply squape/object_tree.py]: Improved in [3bdb214](https://github.com/CyberAlpaca/squish-api-python-extension/pull/52/commits/3bdb21477d598d8af494821bc0a1f0f0decd2ab5)

File: squape/object_tree.py (hunk 2 above)
Comment (pawtom) [reply squape/object_tree.py:297]: I have added debug messages with [397aed9](https://github.com/CyberAlpaca/squish-api-python-extension/pull/52/commits/397aed9392694d2053ced3f00d1868f46becce99). A debug message is provided for each lookup error try.

I don't agree with the statement that if no object is found, you will end up with a lot of lookup errors. I think that in most cases, you will have a single error message per object_name from a list.

File: squape/object_tree.py (hunk 1 above)
Comment (pawtom) [reply squape/object_tree.py]: Improved even more with [633ac11](https://github.com/CyberAlpaca/squish-api-python-extension/pull/52/commits/633ac11ca5395e30b8aa9686f0245825807255c1)

File: squape/object_tree.py (hunk 2 above)
Comment (pawtom) [reply squape/object_tree.py]: Changed with [7a137ac](https://github.com/CyberAlpaca/squish-api-python-extension/pull/52/commits/7a137ac0cfed39855d2e7c0a7cc0d4b6bd28e7c9).
- Debug message with lookup error will be printed for every lookup error (for every try).
- Final error message will not contain lookup errors:
` raise LookupError(f"Objects {object_names} not found")`

**Review by Adakar — APPROVED** [review]
Looks good :)

**Review by Adakar — APPROVED** [review]
👍🏼

**Pull request merged**